demos/http/http_demo_plaintext/http_demo_plaintext.c

```diff
@@ -0,0 +1,553 @@
+
+        LogInfo( ( "Sending HTTP %s request to %s%s...",
+                   pMethod, SERVER_HOST, pPath ) );
+        LogInfo( ( "Request Headers:\n%.*s",
+                   ( int32_t ) requestHeaders.headersLen,
+                   ( char * ) requestHeaders.pBuffer ) );
+        LogInfo( ( "Request Body:\n%.*s\n",
+                   ( int32_t ) REQUEST_BODY_TEXT_LENGTH,
+                   ( char * ) requestBodyBuffer ) );
+        /* Send the request and receive the response. */
+        httpStatus = HTTPClient_Send( pTransport,
+                                      &requestHeaders,
+                                      requestBodyBuffer,
+                                      REQUEST_BODY_TEXT_LENGTH,
+                                      &response,
+                                      0 );
+    }
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        LogInfo( ( "Received HTTP response from %s%s...",
+                   SERVER_HOST, pPath ) );
+        LogInfo( ( "Response Headers:\n%.*s",
+                   ( int32_t ) response.headersLen,
+                   response.pHeaders ) );
+        LogInfo( ( "Response Status:\n%u",
+                   response.statusCode ) );
+        LogInfo( ( "Response Body:\n%.*s\n",
+                   ( int32_t ) response.bodyLen,
+                   response.pBody ) );
+    }
+    else
+    {
+        LogError( ( "Sending HTTP %s request to %s%s failed with status = %u.",
+                    pMethod, SERVER_HOST, pPath, httpStatus ) );
+    }
+
+    return httpStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of demo.
+ */
+int main()
+{
+    int returnStatus = EXIT_SUCCESS;
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+    HTTPNetworkContext_t socketContext = { 0 };
+    HTTPTransportInterface_t transport = { 0 };
+
+    /* Set the request body. */
+    strncpy( ( char * ) requestBodyBuffer,
+             REQUEST_BODY_TEXT, REQUEST_BODY_TEXT_LENGTH );
+
+    /**************************** Connect. ******************************/
+
+    /* Establish TCP connection. */
+    returnStatus = connectToServer( SERVER_HOST, SERVER_PORT, &socketContext.tcpSocket );
+
+    /* Define the transport interface. */
+    transport.recv = transportRecv;
+    transport.send = transportSend;
+    transport.pContext = &socketContext;
+
+    /*********************** Send HTTPS request. ************************/
+
+    /* The client is now connected to the server. This example will send a
+     * GET, HEAD, PUT, and POST request in that order. If any request fails,
+     * an error code is returned. */
+
+    /* Send GET Request. */
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        httpStatus = _sendHttpRequest( &transport,
+                                       SERVER_HOST,
+                                       HTTP_METHOD_GET,
+                                       GET_PATH );
+
+        if( httpStatus != HTTP_SUCCESS )
```

Comment:
This is a little bit weird maybe we can just return EXIT_SUCCESS from _sendHttpRequest? Also because you are not using httpStatus in this function.